charts/quickwit/values.yaml

```diff
@@ -321,6 +321,11 @@ config:
   #
   # Postgres metastore
   # postgres:
+  #   We recommend using an external secret to pass postgres variables to quickwit
+  #   extraEnvFrom:
+  #   - secretRef:
+  #       name: quickwit-postgres
```

Comment:
What do you think about adding an extra attribute `config.postgres.passwordSecretKeyRef` instead (with `name` and `key` fields)?
- one and only one of `passwordSecretKeyRef` or `password` would be required 
- the secret would not be created when `passwordSecretKeyRef` is specified

Reply:
I fully agree with the first comment you described !

> What do you think about adding an extra attribute config.postgres.passwordSecretKeyRef instead (with name and key fields)?
Some users prefer not to commit the hostname used in their Git repository or utilize an operator that retrieves the hostname from an existing secret (to prevent code duplication). Therefore, we should provide options for hostname, port, etc. Perhaps a better approach is to document which fields can currently be populated in both the 'extraEnv' for Quickwit and Metastore, to avoid the necessity of updating the Helm chart each time we support a new option. This approach would enhance the adoption of new features as we wouldn't need to release updates for each change.

> one and only one of passwordSecretKeyRef or password would be required
To not have breaking change, I agree it's what i've did in the current implementation.

> the secret would not be created when passwordSecretKeyRef is specified
Yes, we can. I don't see any issue doing it like this or in the other order.

Reply:
thanks for this detailed response!

> Some users prefer not to commit the hostname

I see, good point. I struggle evaluating whether it is common enough to actually address it right now. My intuition is that being able to source just the Postgres password and the object store access keys should cover most use cases. If someone really needs to do the same for the host/port, we can add that later.

Are you on board with the following proposition?

Add 3 values:
- `config.postgres.password_secret_key_ref`
- `config.storage.s3.secret_access_key_secret_key_ref`
- `config.storage.azure.access_key_secret_key_ref`

With:
- these values are objects with `name` and `key` fields
- only one of `xxx` and `xxx_secret_key_ref` can be defined at any time
- when using `xxx` instead of `xxx_secret_key_ref`, the value is stored as [an internal secret](https://github.com/quickwit-oss/helm-charts/blob/2c6d499a0b0030424c4447a279172d1bf1e71da9/charts/quickwit/templates/secret.yaml)

Reply:
(side note: we can also address https://github.com/quickwit-oss/helm-charts/issues/81 while we are there)

Reply:
Okay, no problem 

> only one of xxx and xxx_secret_key_ref can be defined at any time

I don't know if it's possible to do that with helm, I can prioritize one way over the other but I don't know if you can enforce this.

> (side note: we can also address https://github.com/quickwit-oss/helm-charts/issues/81 while we are there)

Sure, just to be sure I need to add `QW_METASTORE_URI` to `indexer-statefulset.yaml` and `searcher-statefulset.yaml` ?

Reply:
Additionally, I'm not a big fan of only passing the S3 key secret. If you haven't noticed the AWS S3 pricing issue, every request made to S3 incurs a charge (even if it's a forbidden). Though they are working on a fix, it's not functioning yet. Moreover, it's a security concern; the URL and username should be hidden (same for database).

https://medium.com/@maciej.pocwierz/how-an-empty-s3-bucket-can-make-your-aws-bill-explode-934a383cb8b1

Reply:
Okay, i've updated the PR following your proposition.
Few things : 

- After reviewing the helpers.tpl file, I believe non-secret variables should be migrated to a dedicated ConfigMap instead of being passed individually to each container. We could have one ConfigMap for the metastore and another for Quickwit.

- I'm not a big fan of only passing the S3 key secret. If you haven't noticed the AWS S3 pricing issue, every request made to S3 incurs a charge (even if it's a forbidden). Though they are working on a fix, it's not functioning yet. Moreover, it's a security concern; the URL and username should be hidden (same for database).

https://medium.com/@maciej.pocwierz/how-an-empty-s3-bucket-can-make-your-aws-bill-explode-934a383cb8b1

But that's up to you as you are the maintainer here :)

Reply:
> After reviewing the helpers.tpl file, I believe non-secret variables should be migrated to a dedicated ConfigMap instead of being passed individually to each container. We could have one ConfigMap for the metastore and another for Quickwit.

I think this is a good idea! I hate this helpers.tpl, if we could shrink it that would be amazing 😉. Would you be interested in contributing this in a follow up PR?

> If you haven't noticed the AWS S3 pricing issue, every request made to S3 incurs a charge 
For AWS S3, using the environment variables for providing the credentials is already not recommended, users should leverage roles attached to the pods. 

> it's a security concern; the URL and username should be hidden 
I'm torn here. Yes we could hide every parameter. But I don't see a reasonable way of doing this in term of UX and maintainability without introducing a breaking change. I think what you did here already improves security 10x while being easy to use, easy to maintain and retro compatible.

Reply:
> I think this is a good idea! I hate this helpers.tpl, if we could shrink it that would be amazing 😉. Would you be interested in contributing this in a follow up PR?

Yes I can do it.

> For AWS S3, using the environment variables for providing the credentials is already not recommended, users should leverage roles attached to the pods.

AWS S3 were an example not every user use s3, I use rook ceph for example :D